strapi/strapi
# fix: get condigured field for admin_users collection for relation field.

### What does it do?

For the issue that the user was not able to see/select while creating/editing a new entry, the field of the related collection:

I have added a case for checking the read permission of the collection user(admin).   If the user has permission, we can see the corresponding field values, or else the id field is displayed.
 
I have configured the view to firstname for the relation field and am able to see the respective field.

![image](https://github.com/strapi/strapi/assets/58825865/54db75a8-424d-405b-b728-26e5004fb7d2)


### Why is it needed?

I have configured the view to firstname for the relation field. But I am able to see the ID field for the collection while creating an enrty.

![image](https://github.com/strapi/strapi/assets/58825865/a9017510-2cc1-414b-9e77-3e8bce570b75)

This was because while sanitizing the field, we were not checking for the user(admin) collection. So the condition would return id by default.

![image](https://github.com/strapi/strapi/assets/58825865/d62f621c-4cdf-4d19-80b3-ffd651a55b6e)

### How to test it?

- Add a field that is related to the User (from admin) to any content type
- Configure the view for this content type
- Attempt to create an item
- The Relation Fieldcorrect input only shows the ID of the user
- The record is shown correctly in the content manager

### Related issue(s)/PR(s)

Modified the code of the PR -  #17702

closes - #18072
